tests/test_fields.py

```diff
@@ -505,6 +505,47 @@ async def name(self):  # noqa: D102
     assert await page.name == "namex"
 
 
+def test_field_processors_instance() -> None:
+    def proc1(s, instance):
+        return instance.prefix + s + "x"
+
+    @attrs.define
+    class Page(ItemPage):
+        @field(out=[str.strip, proc1])
+        def name(self):  # noqa: D102
+            return "  name\t "
+
+        @field
+        def prefix(self):  # noqa: D102
+            return "prefix: "
+
+    page = Page()
+    assert page.name == "prefix: namex"
+
+
+def test_field_processors_circular() -> None:
+    def proc1(s, instance):
+        return s + instance.b
+
+    def proc2(s, instance):
+        return s + instance.a
+
+    @attrs.define
+    class Page(ItemPage):
+        @field(out=[proc1])
+        def a(self):  # noqa: D102
+            return "a"
+
+        @field(out=[proc2])
+        def b(self):  # noqa: D102
+            return "b"
+
+    page = Page()
+    with pytest.raises(RecursionError):
```

Comment:
We need a better test here (one that doesn't raise) but I haven't thought yet what would be a real example of fields that could reference each other (and how should that work).